dbt/adapters/athena/impl.py

```diff
@@ -208,19 +208,27 @@ def is_work_group_output_location_enforced(self) -> bool:
         else:
             return False
 
-    def _s3_table_prefix(self, s3_data_dir: Optional[str]) -> str:
+    def _s3_table_prefix(
+        self, s3_data_dir: Optional[str], s3_temp_table_dir: Optional[str], is_temporary_table: bool
+    ) -> str:
         """
         Returns the root location for storing tables in S3.
         This is `s3_data_dir`, if set at the model level, the s3_data_dir of the connection if provided,
         and `s3_staging_dir/tables/` if nothing provided as data dir.
         We generate a value here even if `s3_data_dir` is not set,
         since creating a seed table requires a non-default location.
+
+        When `s3_temp_table_dir` is set, we use that as the root location for temporary tables.
         """
         conn = self.connections.get_thread_connection()
         creds = conn.credentials
+
+        s3_temp_table_dir = s3_temp_table_dir or creds.s3_temp_table_dir
+        if s3_temp_table_dir and is_temporary_table:
+            return path.join(s3_temp_table_dir, "temp_tables")
```

Comment:
is this prefix strictly necessary? can't the user pass the prefix that she want in the s3_temp_table_dir configuration?

Reply:
No, the prefix does not required, so I omitted it.
[5453624](https://github.com/dbt-athena/dbt-athena/pull/434/commits/54536248ba37c1235252762afdea15fb35637778)

Reply:
Thanks, doing so we make the user more flexible.